components/schema-migrator/migrations/director/20230223160000_ord_filter_by_formations.up.sql

```diff
@@ -0,0 +1,438 @@
+BEGIN;
+
+-- drop views that rely on 'apps_subaccounts' view so that we can drop and change it
+
+DROP VIEW IF EXISTS tenants_specifications; -- drop tenant_specifications because tenants_apis and tenants_events depend on it. There are no specific changes on this view.
+DROP VIEW IF EXISTS tenants_apis;
+DROP VIEW IF EXISTS tenants_apps;
+DROP VIEW IF EXISTS tenants_bundles;
+DROP VIEW IF EXISTS tenants_events;
+DROP VIEW IF EXISTS tenants_destinations;
+DROP VIEW IF EXISTS tenants_packages;
+DROP VIEW IF EXISTS tenants_products;
+DROP VIEW IF EXISTS tenants_tombstones;
+DROP VIEW IF EXISTS tenants_vendors;
+
+DROP VIEW IF EXISTS apps_subaccounts;
+
+-- enhance the apps_subaccounts view to return not only the ID of the application and the tenant ID (subaccount) in formation with it but also the formation ID
+
+CREATE VIEW apps_subaccounts(id, tenant_id, formation_id)
+AS
+SELECT l.app_id                 AS id,
+       asa.target_tenant_id AS tenant_id,
+       (SELECT id FROM formations f WHERE name = asa.scenario AND (f.tenant_id = asa.tenant_id OR f.tenant_id = asa.target_tenant_id)) as formation_id
```

Comment:
`f.tenant_id = asa.tenant_id OR f.tenant_id = asa.target_tenant_id` why we're checking both `tenant_id` and `target_tenant_id`. Shouldn't the `f.tenant_id = asa.tenant_id` check be enough?

Reply:
On paper it should be enough, but I checked that some formations have subaccount as their tenant on all landscapes:

`select distinct type from business_tenant_mappings where id in (select tenant_id from formations)` returns `account & subaccount`

So, I decided to add both tenants in the query to cover all cases.

Reply:
Hm, most probably these formations are for testing purposes? We should not have productive formations in subaccount? And if the data is for testing purposes maybe we shouldn't adapt the productive code based on that? We could sync it offline as well.

Reply:
Discussed offline - will remove the `f.tenant_id = asa.target_tenant_id` part of the query because it shouldn't happen in a productive flow.